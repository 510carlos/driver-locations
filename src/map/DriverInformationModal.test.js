import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DriverInfromationModal from './DriverInformationModal';

const currentDriver = {
  driverName: 'Carlos Mendoza',
  deliveries: 10,
  remaining: 1,
  nextDestination: '123 fake st',
};

const setModalStatus = () => jest.fn();
const handleCancel = () => jest.fn();

const modal = (
  <DriverInfromationModal
    currentDriver={currentDriver}
    modalStatus
    setModalStatus={setModalStatus}
    handleCancel={handleCancel}
  />
);

it('Mount modal component and take a snapshot', () => {
  const mounted = mount(modal);
  expect(toJSON(mounted)).toMatchSnapshot();
});

test('Mount DriverInformationMount and vrify display information is correct', () => {
  const wrap = mount(modal);
  const driverInfo = wrap.find('ul li');

  const name = driverInfo.at(0).text();
  const deliveries = driverInfo.at(1).text();
  const remaining = driverInfo.at(2).text();
  const next = driverInfo.at(3).text();

  expect(name).toEqual('Driver Name: Carlos Mendoza');
  expect(deliveries).toEqual('Number of deliveries: 10');
  expect(remaining).toEqual('Number of remaining: 1');
  expect(next).toEqual('Next destination: 123 fake st');
});
