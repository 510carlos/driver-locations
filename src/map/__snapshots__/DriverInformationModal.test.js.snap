// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount modal component and take a snapshot 1`] = `
<DriverInformationModal
  currentDriver={
    Object {
      "deliveries": 10,
      "driverName": "Carlos Mendoza",
      "nextDestination": "123 fake st",
      "remaining": 1,
    }
  }
  handleCancel={[Function]}
  modalStatus={true}
  setModalStatus={[Function]}
>
  <Modal
    confirmLoading={false}
    footer={null}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="Driver Information"
    transitionName="zoom"
    visible={true}
    width={520}
  >
    <Dialog
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <ForwardRef(CloseOutlined)
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      footer={null}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      title="Driver Information"
      transitionName="zoom"
      visible={true}
      width={520}
      wrapClassName=""
    >
      <PortalWrapper
        visible={true}
      >
        <ForwardRef
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div
                  class="ant-modal-root"
                >
                  <div
                    class="ant-modal-mask"
                  />
                  <div
                    aria-labelledby="rcDialogTitle0"
                    class="ant-modal-wrap"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal"
                      role="document"
                      style="width: 520px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <button
                          aria-label="Close"
                          class="ant-modal-close"
                          type="button"
                        >
                          <span
                            class="ant-modal-close-x"
                          >
                            <span
                              aria-label="close"
                              class="anticon anticon-close ant-modal-close-icon"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="close"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                />
                              </svg>
                            </span>
                          </span>
                        </button>
                        <div
                          class="ant-modal-header"
                        >
                          <div
                            class="ant-modal-title"
                            id="rcDialogTitle0"
                          >
                            Driver Information
                          </div>
                        </div>
                        <div
                          class="ant-modal-body"
                        >
                          <ul>
                            <li>
                              <b>
                                Driver Name:
                              </b>
                               
                              Carlos Mendoza
                            </li>
                            <li>
                              <b>
                                Number of deliveries:
                              </b>
                               
                              10
                            </li>
                            <li>
                              <b>
                                Number of remaining:
                              </b>
                               
                              1
                            </li>
                            <li>
                              <b>
                                Next destination:
                              </b>
                               
                              123 fake st
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              afterClose={[Function]}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <ForwardRef(CloseOutlined)
                    className="ant-modal-close-icon"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={false}
              footer={null}
              getContainer={[Function]}
              getOpenCount={[Function]}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              switchScrollingEffect={[Function]}
              title="Driver Information"
              transitionName="zoom"
              visible={true}
              width={520}
              wrapClassName=""
            >
              <div
                className="ant-modal-root"
              >
                <Mask
                  motionName="fade"
                  prefixCls="ant-modal"
                  style={
                    Object {
                      "zIndex": undefined,
                    }
                  }
                  visible={true}
                >
                  <CSSMotion
                    key="mask"
                    leavedClassName="ant-modal-mask-hidden"
                    motionName="fade"
                    visible={true}
                  >
                    <DomWrapper>
                      <div
                        className="ant-modal-mask"
                        style={
                          Object {
                            "zIndex": undefined,
                          }
                        }
                      />
                    </DomWrapper>
                  </CSSMotion>
                </Mask>
                <div
                  aria-labelledby="rcDialogTitle0"
                  className="ant-modal-wrap"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                      "zIndex": undefined,
                    }
                  }
                  tabIndex={-1}
                >
                  <Content
                    afterClose={[Function]}
                    ariaId="rcDialogTitle0"
                    closable={true}
                    closeIcon={
                      <span
                        className="ant-modal-close-x"
                      >
                        <ForwardRef(CloseOutlined)
                          className="ant-modal-close-icon"
                        />
                      </span>
                    }
                    confirmLoading={false}
                    destroyOnClose={false}
                    footer={null}
                    getContainer={[Function]}
                    getOpenCount={[Function]}
                    maskTransitionName="fade"
                    motionName="zoom"
                    okType="primary"
                    onCancel={[Function]}
                    onClick={[Function]}
                    onClose={[Function]}
                    onVisibleChanged={[Function]}
                    prefixCls="ant-modal"
                    switchScrollingEffect={[Function]}
                    title="Driver Information"
                    transitionName="zoom"
                    visible={true}
                    width={520}
                    wrapClassName=""
                  >
                    <CSSMotion
                      motionName="zoom"
                      onAppearPrepare={[Function]}
                      onEnterPrepare={[Function]}
                      onVisibleChanged={[Function]}
                      removeOnLeave={false}
                      visible={true}
                    >
                      <DomWrapper>
                        <div
                          className="ant-modal"
                          key="dialog-element"
                          onClick={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 520,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "outline": "none",
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <button
                              aria-label="Close"
                              className="ant-modal-close"
                              onClick={[Function]}
                              type="button"
                            >
                              <span
                                className="ant-modal-close-x"
                              >
                                <ForwardRef(CloseOutlined)
                                  className="ant-modal-close-icon"
                                >
                                  <AntdIcon
                                    className="ant-modal-close-icon"
                                    icon={
                                      Object {
                                        "icon": Object {
                                          "attrs": Object {
                                            "focusable": "false",
                                            "viewBox": "64 64 896 896",
                                          },
                                          "children": Array [
                                            Object {
                                              "attrs": Object {
                                                "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                                              },
                                              "tag": "path",
                                            },
                                          ],
                                          "tag": "svg",
                                        },
                                        "name": "close",
                                        "theme": "outlined",
                                      }
                                    }
                                  >
                                    <span
                                      aria-label="close"
                                      className="anticon anticon-close ant-modal-close-icon"
                                      role="img"
                                    >
                                      <IconReact
                                        className=""
                                        icon={
                                          Object {
                                            "icon": Object {
                                              "attrs": Object {
                                                "focusable": "false",
                                                "viewBox": "64 64 896 896",
                                              },
                                              "children": Array [
                                                Object {
                                                  "attrs": Object {
                                                    "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                                                  },
                                                  "tag": "path",
                                                },
                                              ],
                                              "tag": "svg",
                                            },
                                            "name": "close",
                                            "theme": "outlined",
                                          }
                                        }
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="close"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          key="svg-close"
                                          viewBox="64 64 896 896"
                                          width="1em"
                                        >
                                          <path
                                            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                            key="svg-close-svg-0"
                                          />
                                        </svg>
                                      </IconReact>
                                    </span>
                                  </AntdIcon>
                                </ForwardRef(CloseOutlined)>
                              </span>
                            </button>
                            <div
                              className="ant-modal-header"
                            >
                              <div
                                className="ant-modal-title"
                                id="rcDialogTitle0"
                              >
                                Driver Information
                              </div>
                            </div>
                            <div
                              className="ant-modal-body"
                            >
                              <ul>
                                <li>
                                  <b>
                                    Driver Name:
                                  </b>
                                   
                                  Carlos Mendoza
                                </li>
                                <li>
                                  <b>
                                    Number of deliveries:
                                  </b>
                                   
                                  10
                                </li>
                                <li>
                                  <b>
                                    Number of remaining:
                                  </b>
                                   
                                  1
                                </li>
                                <li>
                                  <b>
                                    Next destination:
                                  </b>
                                   
                                  123 fake st
                                </li>
                              </ul>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "outline": "none",
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </DomWrapper>
                    </CSSMotion>
                  </Content>
                </div>
              </div>
            </Dialog>
          </Portal>
        </ForwardRef>
      </PortalWrapper>
    </Dialog>
  </Modal>
</DriverInformationModal>
`;
